tests/ember_debug/view-debug-test.js

```diff
@@ -1,44 +1,93 @@
 import {
   click,
   find,
-  settled,
   triggerEvent,
   visit
 } from '@ember/test-helpers';
+import hasEmberVersion from '@ember/test-helpers/has-ember-version';
 import { A } from '@ember/array';
 import { run } from '@ember/runloop';
-import Component from '@ember/component';
-import Route from '@ember/routing/route';
+import EmberComponent from '@ember/component';
+import EmberRoute from '@ember/routing/route';
 import EmberObject from '@ember/object';
 import Controller from '@ember/controller';
-import { inspect } from '@ember/debug';
-import { module, skip, test } from 'qunit';
+import { module, test } from 'qunit';
 import { hbs } from 'ember-cli-htmlbars';
 import require from 'require';
+import QUnit from 'qunit';
 import { destroyEIApp, setupEIApp } from '../helpers/setup-destroy-ei-app';
-import hasEmberVersion from '@ember/test-helpers/has-ember-version';
 
 const EmberDebug = require('ember-debug/main').default;
 let port;
 let App;
 
+// TODO make the debounce configurable for tests
+async function timeout(ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms);
+  });
+}
+
+// TODO switch to an adapter architecture, similar to the acceptance tests
+async function captureMessage(type, callback) {
+  if (!port) {
+    throw new Error('Cannot call captureMessage without a port');
+  }
+
+  let send = port.send;
+
+  try {
+    let captured;
+
+    port.send = (name, message) => {
+      if (!captured && name === type) {
+        captured = message;
+      } else {
+        send.call(port, name, message);
+      }
+    };
+
+    await callback();
+
+    if (captured) {
+      return captured;
+    } else {
+      throw new Error(`Did not send a message of type ${type}`);
+    }
+  } finally {
+    port.send = send;
+  }
+}
+
+async function getRenderTree() {
+  let message = await captureMessage('view:renderTree', async () => {
+    port.trigger('view:getTree');
+    await timeout(300);
+  });
+
+  if (message) {
+    return message.tree;
+  }
+}
+
 function setTemplate(name, factory) {
   if (typeof factory.meta === 'object') {
-    factory.meta.moduleName = `my-app/${name}.hbs`;
+    factory.meta.moduleName = `my-app/templates/${name}.hbs`;
   } else if (typeof factory.__meta === 'object') {
     // Ember 3.13+
-    factory.__meta.moduleName = `my-app/${name}.hbs`;
+    factory.__meta.moduleName = `my-app/templates/${name}.hbs`;
   }
```

Comment:
It isn't immediately obvious to me why we actually care about setting `meta.moduleName`, what is it used for in the UI?

Reply:
We use it to show the template name/path in the tooltips